pymc3/progressbar.py

```diff
@@ -1,140 +0,0 @@
-"""
-A simple progress bar to monitor MCMC sampling progress.
-Modified from original code by Corey Goldberg (2010)
-"""
-
-from __future__ import print_function
-
-import warnings
-warnings.simplefilter(action="ignore", category=FutureWarning)
-import sys
-import time
-import datetime
-import uuid
-try:
-    __IPYTHON__
-    from IPython.core.display import HTML, Javascript, display
-except (NameError, ImportError):
-    pass
-
-__all__ = ['progress_bar']
-
-
-class ProgressBar(object):
-
-    def __init__(self, iterations, animation_interval=.5):
-        self.iterations = iterations
-        self.start = time.time()
-        self.last = 0
-        self.animation_interval = animation_interval
-
-    def percentage(self, i):
-        return 100 * i / float(self.iterations)
-
-    def update(self, i):
-        elapsed = time.time() - self.start
-        i += 1
-
-        if elapsed - self.last > self.animation_interval:
-            self.animate(i, elapsed)
-            self.last = elapsed
-        elif i == self.iterations and i != 1:
-            self.animate(i, elapsed)
-
-
-class TextProgressBar(ProgressBar):
-
-    def __init__(self, iterations, printer):
-        self.fill_char = '-'
-        self.width = 20
-        self.printer = printer
-
-        super(TextProgressBar, self).__init__(iterations)
-        self.update(0)
-
-    def animate(self, i, elapsed):
-        self.printer(self.progbar(i, elapsed))
-
-    def progbar(self, i, elapsed):
-        bar = self.bar(self.percentage(i))
-        sps = i / float(elapsed)
-        eta = (self.iterations / sps) - elapsed
-        its = self.iterations
-
-        prog_str = "[{bar}] {it} of {its} in {s_elapsed} sec. " \
```

Comment:
Yeah!  Some sample output:

```
test_disaster_model (pymc3.tests.test_examples.TestDisasterModel) ... 
Applied log-transform to early_mean and added transformed early_mean_log_ to model.
Applied log-transform to late_mean and added transformed late_mean_log_ to model.
Assigned Metropolis to switchpoint
 32%|███████████████████████▎                                                 | 15969/50000 [00:15<00:33, 1016.85it/s]

```

in the brackets at the end are `[time spent < ETA, SPS it/s]`.  This can all be customized.  I did manual checks with examples in a notebook, ipython shell, and from the command line, with similar results.  There are fancy jupyter notebook widgets that can be used, but the base functionality is fine.